### PR TITLE
fix: [CDS-53379]: Show failed resources in logs as well when dry run fails

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
The current library only outputs the resources that are configured successfully. In this PR i have added code to show resources which were unable to be configured also as part of output.

## Testing
Screenshot 2023-03-01 at 11 16 12 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java [hunk 1]
```diff
@@ -1194,12 +1196,25 @@ public boolean dryRunManifests(Kubectl client, List<KubernetesResource> resource
           : overriddenClient.apply().filename("manifests-dry-run.yaml").dryrun(true);
       ProcessResponse response = runK8sExecutable(k8sDelegateTaskParams, executionLogCallback, dryrun);
       ProcessResult result = response.getProcessResult();
+      // Getting Version for the kubectl
+      VersionCommandShort versionCommand = client.versionShort();
+      ProcessResponse versionResponse = runK8sExecutable(k8sDelegateTaskParams, executionLogCallback, versionCommand);
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java:1201]: We shouldn't log anything in executionLogCallback for get version, you can use runK8sExecutableSilent
And let's move the get version logic to different method, please consider:
1. Exit value is not 0 => return empty string
2. processResult.hasOutput() is false => return empty string
3. Use processResult.outputUTF8() instead

File: 960-api-services/src/main/java/io/harness/k8s/kubectl/VersionCommandShort.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.k8s.kubectl;
+
+public class VersionCommandShort extends AbstractExecutable {
```
Comment (abosii-harness) [inline 960-api-services/src/main/java/io/harness/k8s/kubectl/VersionCommandShort.java:10]: let's reuse VersionCommand as discussed
And let's see if we can use --output=json instead

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 960-api-services/src/main/java/io/harness/k8s/kubectl/VersionCommandShort.java (hunk 2 above)
Comment (TarunUba1) [reply 960-api-services/src/main/java/io/harness/k8s/kubectl/VersionCommandShort.java]: Done

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java (hunk 1 above)
Comment (TarunUba1) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java]: Done

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 5 of 10 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)* status: 11 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 1207 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRWlh1BnLNE3-_f2Pg:-NPRWlh1BnLNE3-_f2Ph:bqffa9c) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L1207)):*
> ```Java
>             resource.getResourceId().getKind().toLowerCase() + "/" + resource.getResourceId().getName();
>         if (!(resourcesCreated.contains(resourceName))) {
>           resourcesNotCreatedBuilder.append(resourceName + " \n");
> ```

.append(resourceName).append("\n")
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 3299 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRW0aB2EDXsl2B9ayC:-NPRW0aB2EDXsl2B9ayD:b-plegee) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L3299)):*
> ```Java
>   }
>   @VisibleForTesting
>   public ProcessResult getKubernetesVersion(K8sDelegateTaskParams k8sDelegateTaskParams, Kubectl client)
> ```

This method should return String and not process result. If it's a failure then it should be an empty string
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 3310 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRVvnX79uXePkvOqF-:-NPRVvnX79uXePkvOqF0:b18cd4p) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L3310)):*
> ```Java
>       return new ProcessResult(0, new ProcessOutput("".getBytes(UTF_8)));
>     }
>     JSONObject jsonObject = new JSONObject(versionResult.outputUTF8());
> ```

Can we move this logic to exception handler?
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRXB4R5SqGzPbHFtR0:-NPRXB4R5SqGzPbHFtR1:b-h2hdzf) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L21)):*
> ```Java
>         hintMessage, new ExplanationException(explanationMessage, new ExplanationException(commandExecuted, cause)));
>   }
>   public static WingsException hintWithExplanationAndCommandExceptionWithExtraInfo(String hintMessage,
> ```

Either move this to k8s exception handler or generalize it with (hint, explanation, cause, explanations...) and then for each explanation in var args wrap final exception in new ExplanationException

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)* status: 11 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 1207 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRWlh1BnLNE3-_f2Pg:-NPX5Snv2XJzathVEipV:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L1207)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

.append(resourceName).append("\n")
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 3299 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRW0aB2EDXsl2B9ayC:-NPX5UEaELG1ybt7s9Jj:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L3299)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

This method should return String and not process result. If it's a failure then it should be an empty string
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 3310 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRVvnX79uXePkvOqF-:-NPX5V-Y0sHPdPZ4Aish:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L3310)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Can we move this logic to exception handler?
</blockquote></details>

Done.
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRXB4R5SqGzPbHFtR0:-NPX5Wat6Xe2c7ITOqlS:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L21)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Either move this to k8s exception handler or generalize it with (hint, explanation, cause, explanations...) and then for each explanation in var args wrap final exception in new ExplanationException
</blockquote></details>

Done.

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 5 files at r3, 4 of 5 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)* status: 11 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRXB4R5SqGzPbHFtR0:-NPXIOKNEG5ZvtLCErJT:b-k7q4sl) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L21)):*
<details><summary><i>Previously, TarunUba1 (Tarun Uba) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

please rename this to hintWithExplanations
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 23 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXIb_L3kXkQtEZOD64:-NPXIb_L3kXkQtEZOD65:b-lylyam) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L23)):*
> ```Java
>   public static WingsException hintWithExplanationAndCommandExceptionWithExtraInfo(
>       String hintMessage, Throwable cause, String... explanations) {
>     int numberOfExplanations = explanations.length;
> ```

We will need to have a check if length is > 0

**Review by abhinavsingh1196 — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)* status: 11 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 1210 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXFqf87-8zRRGKnsRu:-NPXFqf87-8zRRGKnsRv:bzf3jbk) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L1210)):*
> ```Java
>       resources.forEach(resource -> {
>         String resourceName =
>             resource.getResourceId().getKind().toLowerCase() + "/" + resource.getResourceId().getName();
> ```

io.harness.k8s.model.KubernetesResourceId#kindNameRef performs the same function, let's use that. 

also, while checking below condition `!(resourcesCreated.contains(resourceName))` do we need to have both as lowercase, just to be safe? 
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/exception/KubernetesCliRuntimeExceptionHandler.java` line 160 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXHH0i3Qocn-ervGeH:-NPXHH0i3Qocn-ervGeI:b-i1m6yw) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/exception/KubernetesCliRuntimeExceptionHandler.java#L160)):*
> ```Java
>       List<String> extractedValues = extractValuesFromFirstMatch(cliErrorMessage, INVALID_TYPE_ERROR_PATTERN, 1);
>       String invalidFieldName = extractedValues.isEmpty() ? "" : extractedValues.get(0);
>       String resourcesNotApplied = (isEmpty(kubernetesTaskException.getResourcesNotApplied()))
> ```

redundant braces
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/exception/KubernetesCliRuntimeExceptionHandler.java` line 164 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXHLdE34c8ekhPP3Uk:-NPXHLdE34c8ekhPP3Ul:b-3kfjsu) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/exception/KubernetesCliRuntimeExceptionHandler.java#L164)):*
> ```Java
>           : kubernetesTaskException.getResourcesNotApplied();
>       String kubectlVersion = EMPTY;
>       if (!isEmpty(kubernetesTaskException.getKubectlVersion())) {
> ```

use isNotEmpty
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 21 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXHp8C3rpPXApV42hR:-NPXHp8C3rpPXApV42hS:b-d3ogzk) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L21)):*
> ```Java
>         hintMessage, new ExplanationException(explanationMessage, new ExplanationException(commandExecuted, cause)));
>   }
>   public static WingsException hintWithExplanationAndCommandExceptionWithExtraInfo(
> ```

can rename to `hintWithExplanationsException`
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 23 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXIJ5_DA0mWO1BQcCO:-NPXIJ5_DA0mWO1BQcCP:b-dihxqr) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L23)):*
> ```Java
>   public static WingsException hintWithExplanationAndCommandExceptionWithExtraInfo(
>       String hintMessage, Throwable cause, String... explanations) {
>     int numberOfExplanations = explanations.length;
> ```

add edge cases (not applicable now, but future proofing)

if (numberOfExplanations == 0) return new HintException(hintMessage, cause);

if (numberOfExplanations == 1) return hintWithExplanationException(hintMessage, explanations[0], cause);

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)* status: 11 file(s) to review. 2 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 1210 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXFqf87-8zRRGKnsRu:-NPXPlVuEMXOd-b1FdC4:b-m8u5sn) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L1210)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

io.harness.k8s.model.KubernetesResourceId#kindNameRef performs the same function, let's use that. 

also, while checking below condition `!(resourcesCreated.contains(resourceName))` do we need to have both as lowercase, just to be safe? 
</blockquote></details>

Lowercase not needed for the resourcename
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/exception/KubernetesCliRuntimeExceptionHandler.java` line 160 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXHH0i3Qocn-ervGeH:-NPXPyuE8CrWNYEZhoOp:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/exception/KubernetesCliRuntimeExceptionHandler.java#L160)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

redundant braces
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/exception/KubernetesCliRuntimeExceptionHandler.java` line 164 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXHLdE34c8ekhPP3Uk:-NPXPzS81tso4MLbcWnt:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/exception/KubernetesCliRuntimeExceptionHandler.java#L164)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

use isNotEmpty
</blockquote></details>

Done.
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/44389#-NPRXB4R5SqGzPbHFtR0:-NPXQ1Sl4Wn_SYmfaOJm:b-e22php) ([raw file](https://github.com/harness/harness-core/blob/d417b3da800377c1a5ddac78277f7fea00fc32b0/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L21)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

please rename this to hintWithExplanations
</blockquote></details>

used abhis
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 21 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXHp8C3rpPXApV42hR:-NPXQ-8k6ZCaAzYCzdV2:b-don2ke) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L21)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

can rename to `hintWithExplanationsException`
</blockquote></details>

used yours
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 23 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXIb_L3kXkQtEZOD64:-NPXQ3-e8P642JKSMrBw:bdqgn5j) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L23)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

We will need to have a check if length is > 0
</blockquote></details>

checking if 0 then returning otherwise works
___
*[`980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java` line 23 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXIJ5_DA0mWO1BQcCO:-NPXQ8Ml7L9OfvH5KtKh:bs2dhe1) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/980-commons/src/main/java/io/harness/exception/NestedExceptionUtils.java#L23)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

add edge cases (not applicable now, but future proofing)

if (numberOfExplanations == 0) return new HintException(hintMessage, cause);

if (numberOfExplanations == 1) return hintWithExplanationException(hintMessage, explanations[0], cause);
</blockquote></details>

only checking case 0 as for 1 the code works

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)* status: 11 file(s) to review. 2 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 1210 at r5](https://reviewable.io/reviews/harness/harness-core/44389#-NPXFqf87-8zRRGKnsRu:-NPXmG9KDPrQfDeq0EYF:b-896fjx) ([raw file](https://github.com/harness/harness-core/blob/cb01b893c475006d4e2ccf711332f7cd146b4ada/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L1210)):*
<details><summary><i>Previously, TarunUba1 (Tarun Uba) wrote…</i></summary><blockquote>

Lowercase not needed for the resourcename
</blockquote></details>

Done

**Review by namantalaycha10 — APPROVED** [review]
> Reviewed 1 of 10 files at r1, 4 of 6 files at r2, 1 of 5 files at r3, 2 of 5 files at r4, 2 of 3 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)* status: 1 change requests. 5. unresolved discussions.

**Review by abhinavsingh1196 — APPROVED** [review]
> Reviewed 1 of 10 files at r1, 4 of 6 files at r2, 1 of 5 files at r3, 2 of 5 files at r4, 2 of 3 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44389)* status: :shipit: complete! ✓

**Pull request merged**